.cursor/rules/tools/fetch_pr_diff.py

```
@@ -0,0 +1,115 @@
+
+def get_pr_number():
+    """Fetches the current PR number if available."""
+    # Try multiple approaches to get PR number
+    
+    # Method 1: Try with explicit shell=False and full path
+    try:
+        result = subprocess.run(["/opt/homebrew/bin/gh", "pr", "view", "--json", "number", "--jq", ".number"], 
+                              capture_output=True, text=True, shell=False)
+        if result.returncode == 0 and result.stdout.strip():
+            return result.stdout.strip()
+    except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)